### PR TITLE
doc: create release guide

### PR DESCRIPTION
**Pull request by MylesBorins**
This is a rough first pass. Comments and suggestions welcome :D

### CODE REVIEW

File: RELEASES.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+```bash
+$ semver-sync -t patch # for patch
+$ semver-sync -t minor # for minor
+$ semver-sync -t major # for major
+```
+
+4. Push to github
+
+```bash
+$ git push origin master
```
Comment (targos) [inline RELEASES.md]: Shouldn't it be `upstream` ?
The new tag can be pushed in the same command with `--follow-tags` option

File: RELEASES.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+Any member of the @nodejs/citgm team can make a release.
+
+## How to create a release
+
+1. Ensure your branch is up to date with upstream
+
+```bash
+$ git remote update -p
+$ git rebase upstream/master
```
Comment (gibfahn) [inline RELEASES.md]: What branch is this? I'd assume it should always be master, in which case I'd normally do:

```bash
git checkout master
git fetch upstream # or git fetch --all
git reset --hard upstream/master
```

to be clear about what branch you should be on.

---

After a quick look I'm not sure what the difference is between `git remote update -p` and `git fetch --all -p` (or why you'd bother with the `--prune`. I'm more familiar with `fetch`, but it's NBD either way.

File: RELEASES.md (hunk 1 above)
Comment (MylesBorins) [inline RELEASES.md]: updated

File: RELEASES.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+```bash
+$ npm install -g semver-sync
+```
+
+3. Bump the version and create tag with semver-sync
+
+```bash
+$ semver-sync -t patch # for patch
+$ semver-sync -t minor # for minor
+$ semver-sync -t major # for major
```
Comment (gibfahn) [inline RELEASES.md]: Is there a reason to use this and not just use `npm version patch|minor|major`? The preversion and postversion scripts look pretty useful, and it removes a dependency.

```json
"scripts": {
  "preversion": "npm test",
  "version": "npm run build && git add -A dist",
  "postversion": "git push && git push --tags && rm -rf build/temp"
}
```

Comment (MylesBorins) [inline RELEASES.md]: so we could use `npm version` instead, which also tags + creates a commit... it is just a slightly different format than we have done up until now for the commit message... I'm ok with this though. Not a fan of the postversion  auto push

**Comment by MylesBorins** [comment]
updated based on feedback ptal

File: RELEASES.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
+# CITGM Release Process
+
+This document describes the technical aspects of the citgm release process. The intended audience is those who have been authorized to create official releases for citgm, hosted on npm.
+
+## Who can make a release?
+
+Any member of the @nodejs/citgm team can make a release.
```
Comment (gibfahn) [inline RELEASES.md]: Is this true? I'd assume you have to be added as a maintainer on npm.

File: RELEASES.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
+# CITGM Release Process
+
+This document describes the technical aspects of the citgm release process. The intended audience is those who have been authorized to create official releases for citgm, hosted on npm.
```
Comment (gibfahn) [inline RELEASES.md]: >The intended audience is those who have been authorized to create official releases for citgm, hosted on npm.

Is this `@nodejs/citgm` as below? If so the whole sentence seems a bit redundant.

File: RELEASES.md (hunk 4 above)
Comment (MylesBorins) [inline RELEASES.md]: I was imagining that with this language we would begin giving the publish bit to anyone who is part of @nodejs/citgm

File: RELEASES.md (hunk 5 above)
Comment (MylesBorins) [inline RELEASES.md]: it is, this was lifted from Release doc on main repo

File: RELEASES.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+$ npm version [<newversion> | major | minor | patch]
+```
+
+3. Push to github
+
+```bash
+$ git push upstream master --follow-tags
+```
+
+6. Publish to npm
```
Comment (targos) [inline RELEASES.md]: 4.

Comment (MylesBorins) [inline RELEASES.md]: fixed

**Pull request merged**